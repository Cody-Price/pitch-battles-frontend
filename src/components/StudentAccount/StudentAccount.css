.student-account-page {
  width: 700px;
  height: 332px;
  margin: 3rem auto;
  padding: 1rem;
  background: url(./images/student-account-frame.png);
  background-size: cover;
  image-rendering: pixelated;
  position: relative;
  right: 0;
  z-index: 200;
  display: grid;
  grid-template-columns: 200px 1fr;
  grid-template-rows: 50px 1fr;
  font-family: "chronotype";
  text-align: start;
  animation: slide-in-student-account 0.3s linear;
  -webkit-animation: slide-in-student-account 0.3s linear;
  -moz-animation: slide-in-student-account 0.3s linear;
  -o-animation: slide-in-student-account 0.3s linear;
}

.student-account-page ul {
  padding: 0;
  margin: 1rem;
  padding-left: 1rem;
}

.student-account-page li {
  list-style: none;
  padding: 0.3rem;
  margin-bottom: 4px;
  box-shadow: 1px 1px 1px rgba(0, 0, 0, 0);
  transition: 0.3s;
}

.student-account-page li:hover {
  cursor: pointer;
}

.student-account-label {
  padding: 0.5rem;
  margin-left: 1.5rem;
  grid-column: 1/3;
}

.student-account-active-tab {
  padding-left: 2rem;
  grid-column: 2/3;
}

.back-to-dash-button {
  position: absolute;
  right: 2rem;
  top: 2rem;
  background: none;
  border: 0;
  font-family: "chronotype";
  font-size: 1.1rem;
}

.back-to-dash-button:hover {
  cursor: pointer;
}

.profile-link.edit-profile-active,
.avatar-link.edit-avatar-active,
.password-link.change-password-active {
  background: #6a9dd4;
  box-shadow: 1px 1px 1px rgba(0, 0, 0, 0.7);
}

@keyframes slide-in-student-account {
  0% {
    right: 3000px;
  }
  100% {
    right: 0;
  }
}

@-webkit-keyframes slide-in-student-account {
  0% {
    right: 3000px;
  }
  100% {
    right: 0;
  }
}

@-moz-keyframes slide-in-student-account {
  0% {
    right: 3000px;
  }
  100% {
    right: 0;
  }
}

@-o-keyframes slide-in-student-account {
  0% {
    right: 3000px;
  }
  100% {
    right: 0;
  }
}

@media (max-width: 770px) {
  .student-account-page {
    width: 350px;
    height: 570px;
    background: url("../StudentDash/images/mq-stud-dash.png");
    background-size: cover;
    image-rendering: pixelated;
    grid-template-columns: 164px 1fr;
    grid-template-rows: 50px 1fr;
    padding-left: 0;
    margin-top: 0;
    top: 1rem;
  }

  .student-account-label {
    padding: 0;
    margin-left: 3rem;
  }

  ul.student-account-nav {
    padding-left: 0;
    margin-left: 1.7rem;
  }

  .student-account-active-tab {
    padding-left: 0;
  }
}
