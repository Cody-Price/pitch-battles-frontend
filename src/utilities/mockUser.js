const mockUser = {
  first_name: "Chet",
  last_name: "Manly",
  avatar: 10,
  role: 0,
  email: "cmanly@gmail.com",
  games_played: 100,
  class: "6th-grade band"
};

export default mockUser;
