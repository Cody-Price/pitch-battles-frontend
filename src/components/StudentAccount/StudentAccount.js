import React, { Component } from "react";
import "./StudentAccount.css";

import EditAvatar from "../EditAvatar/EditAvatar";
import EditProfile from "../EditProfile/EditProfile";
import ChangePassword from "../ChangePassword/ChangePassword";

class StudentAccount extends Component {
  constructor() {
    super();

    this.state = {
      activePage: "edit-profile-active"
    };
  }

  navigate = string => {
    if (string !== this.state.activePage) {
      this.setState({
        activePage: string
      });
    } else {
      return;
    }
  };

  render() {
    return (
      <main className="student-account-page">
        <button
          onClick={() => {
            this.props.navigate("student dash");
          }}
          className="back-to-dash-button"
        >
          back
        </button>
        <h3 className="student-account-label">user account settings</h3>
        <ul className="student-account-nav">
          <li
            className={`profile-link ${this.state.activePage}`}
            onClick={() => {
              this.navigate("edit-profile-active");
            }}
          >
            edit profile
          </li>
          <li
            className={`avatar-link ${this.state.activePage}`}
            onClick={() => {
              this.navigate("edit-avatar-active");
            }}
          >
            edit avatar
          </li>
          <li
            className={`password-link ${this.state.activePage}`}
            onClick={() => {
              this.navigate("change-password-active");
            }}
          >
            change password
          </li>
        </ul>
        <section className="student-account-active-tab">
          {this.state.activePage === "edit-profile-active" && (
            <EditProfile
              user={this.props.user}
              changeProfile={this.props.changeProfile}
            />
          )}
          {this.state.activePage === "edit-avatar-active" && (
            <EditAvatar
              user={this.props.user}
              changeAvatar={this.props.changeAvatar}
            />
          )}
          {this.state.activePage === "change-password-active" && (
            <ChangePassword
              user={this.props.user}
              changePassword={this.props.changePassword}
            />
          )}
        </section>
      </main>
    );
  }
}

export default StudentAccount;
