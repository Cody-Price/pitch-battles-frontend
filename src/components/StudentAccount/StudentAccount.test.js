import React from "react";

import StudentAccount from "./StudentAccount";
import Enzyme, { shallow } from "enzyme";
import mockUser from "../../utilities/mockUser";

const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

jest.mock("../../utilities/fetchCalls");

describe("StudentAccount", () => {
  let wrapper;
  let mockNavigate;
  let mockLogout;

  beforeEach(() => {
    mockNavigate = jest.fn();
    const mockChangeProfile = jest.fn();
    const mockChangeAvatar = jest.fn();
    const mockChangePassword = jest.fn();
    mockLogout = jest.fn();

    wrapper = shallow(
      <StudentAccount
        navigate={mockNavigate}
        changeProfile={mockChangeProfile}
        changeAvatar={mockChangeAvatar}
        changePassword={mockChangePassword}
        user={mockUser}
        logout={mockLogout}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call navigate on click with the correct params", () => {
    const expected = "student dash";

    wrapper.instance().forceUpdate();

    wrapper.find(".back-to-dash-button").simulate("click");

    expect(mockNavigate).toHaveBeenCalledWith(expected);
  });

  describe("navigate", () => {
    it("should setState if the current page is not equal to the selected page", () => {
      const expected = "edit-avatar-active";

      wrapper.instance().navigate("edit-avatar-active");

      expect(wrapper.state().activePage).toEqual(expected);
    });

    it("should return if the current and selected page are the same", () => {
      const spy = jest.spyOn(wrapper.instance(), "navigate");

      wrapper.instance().navigate("edit-profile-active");

      expect(spy).toReturn();
    });

    it("should call navigate on click with the correct params", () => {
      const spy = jest.spyOn(wrapper.instance(), "navigate");
      const expected = "change-password-active";

      wrapper.instance().forceUpdate();

      wrapper.find(".password-link").simulate("click");

      expect(spy).toHaveBeenCalledWith(expected);
    });

    it("should call navigate on click with the correct params", () => {
      const spy = jest.spyOn(wrapper.instance(), "navigate");
      const expected = "edit-profile-active";

      wrapper.instance().forceUpdate();

      wrapper.find(".profile-link").simulate("click");

      expect(spy).toHaveBeenCalledWith(expected);
    });

    it("should call navigate on click with the correct params", () => {
      const spy = jest.spyOn(wrapper.instance(), "navigate");
      const expected = "edit-avatar-active";

      wrapper.instance().forceUpdate();

      wrapper.find(".avatar-link").simulate("click");

      expect(spy).toHaveBeenCalledWith(expected);
    });
  });

  describe("logout", () => {
    it("should call logout on click", () => {
      wrapper.find(".logout-link").simulate("click");

      expect(mockLogout).toHaveBeenCalled();
    });
  });
});
