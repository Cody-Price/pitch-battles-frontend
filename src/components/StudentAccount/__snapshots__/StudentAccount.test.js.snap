// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentAccount should match the snapshot 1`] = `
<main
  className="student-account-page"
>
  <button
    className="back-to-dash-button"
    onClick={[Function]}
  >
    back
  </button>
  <h3
    className="student-account-label"
  >
    user account settings
  </h3>
  <ul
    className="student-account-nav"
  >
    <li
      className="profile-link edit-profile-active"
      onClick={[Function]}
    >
      edit profile
    </li>
    <li
      className="avatar-link edit-profile-active"
      onClick={[Function]}
    >
      edit avatar
    </li>
    <li
      className="password-link edit-profile-active"
      onClick={[Function]}
    >
      change password
    </li>
    <li
      className="logout-link"
      onClick={[MockFunction]}
    >
      logout
    </li>
  </ul>
  <section
    className="student-account-active-tab"
  >
    <EditProfile
      changeProfile={[MockFunction]}
      user={
        Object {
          "attributes": Object {
            "badges": Object {
              "data": Array [
                Object {
                  "level_one": true,
                },
                Object {
                  "level_one_perfect": true,
                },
                Object {
                  "level_two": true,
                },
                Object {
                  "level_two_perfect": true,
                },
                Object {
                  "level_three": true,
                },
                Object {
                  "level_three_perfect": false,
                },
                Object {
                  "level_four": true,
                },
                Object {
                  "level_four_perfect": false,
                },
                Object {
                  "all_perfect": false,
                },
                Object {
                  "five_games": true,
                },
                Object {
                  "ten_games": true,
                },
                Object {
                  "twenty_games": true,
                },
                Object {
                  "fifty_games": true,
                },
                Object {
                  "one_hundred_games": true,
                },
                Object {
                  "two_hundred_games": false,
                },
                Object {
                  "five_hundred_games": false,
                },
                Object {
                  "one_thousand_games": false,
                },
              ],
            },
            "fastest_times": Object {
              "data": Array [
                Object {
                  "level_one": 11.01,
                },
                Object {
                  "level_two": 15.24,
                },
                Object {
                  "level_three": 24.64,
                },
                Object {
                  "level_four": 30.17,
                },
                Object {
                  "all": 83.16,
                },
              ],
            },
            "games_played": Object {
              "data": 115,
            },
          },
          "avatar": 9,
          "class": "6th-grade band",
          "email": "cmanly@gmail.com",
          "first_name": "Chet",
          "last_name": "Stanley",
          "role": 0,
        }
      }
    />
  </section>
</main>
`;
